atomicapp/nulecule/config.py

```diff
@@ -0,0 +1,176 @@
+
+    def get(self, key):
+        """
+        Get value for a key from data accessible from the current namespace.
+
+        TODO: Improved data inheritance model. It makes sense for a component
+        to be able to access data from it's sibling namespaces and children
+        namespaces.
+
+        Args:
+            key (str): Key
+
+        Returns:
+            Value for the key, else None.
+        """
+        return (
+            self._data[self.namespace].get(key) or
+            (self._data[self._parent_ns].get(key) if self._parent_ns else None) or
+            self._data[GLOBAL_CONF].get(key) or
+            self._answers[self.namespace].get(key) or
+            (self._answers[self._parent_ns].get(key) if self._parent_ns else None) or
+            self._answers[GLOBAL_CONF].get(key)
+        )
+
+    def context(self):
+        """
+        Get context to render artifact files in a Nulecule component.
+
+        TODO: Improved data inheritance model. Data from siblings and children
+        namespaces should be available in the context to render an artifact
+        file in the current namespace.
+        """
+        if self._context is None:
+            self._context = {}
+            self._context.update(copy.copy(self._data[GLOBAL_CONF]))
+            self._context.update(copy.copy(self._data[self.namespace]))
+
+            self._context.update(copy.copy(self._answers[GLOBAL_CONF]))
+            self._context.update(copy.copy(self._answers[self.namespace]))
+        return self._context
+
+    def runtime_answers(self):
+        """
+        Get runtime answers.
+
+        Returns:
+            A defaultdict containing runtime answers data.
+        """
+        answers = defaultdict(dict)
+        answers.update(copy.copy(self._answers))
+
+        for key, value in self._data.items():
+            answers[key].update(value)
+
+        # Remove empty sections for answers
+        for key, value in answers.items():
+            if value is None:
+                answers.pop(key, None)
+
+        return answers
+
+    def clone(self, namespace):
+        """
+        Create a new config instance in the specified namespace.
+
+        Args:
+            name (str): Name of the child component
+
+        Returns:
+            A Config instance.
+        """
+        config = Config(namespace=namespace,
+                        answers=self._answers,
+                        cli=self._cli,
+                        data=self._data)
+        return config
+
+    def _split_namespace(self, namespace):
+        """
+        Split namespace to get parent and current namespace in a Nulecule.
+        """
+        if self._is_nulecule:
+            return '', namespace
+        words = namespace.rsplit(NAMESPACE_SEPARATOR, 1)
+        parent, current = '', ''
+        if len(words) == 2:
+            parent, current = words[0], words[1]
+        else:
+            parent, current = '', words[0]
+        return parent, current
```

Comment:
Bit confused with this function. What's up with the separating?

What happens if there is a space in the namespace?

What are we checking here?